src/workerd/io/worker.c++

```diff
@@ -448,37 +448,105 @@ public:
       // We're on a request-serving thread.
       auto& ioContext = IoContext::current();
       timePoint = ioContext.now();
-    } else KJ_IF_MAYBE(info, inspectorTimerInfo) {
-      if (info->threadId == getCurrentThreadId()) {
-        // We're on an inspector-serving thread.
+    } else {
+      auto lockedState = state.lockExclusive();
+      KJ_IF_MAYBE(info, lockedState->inspectorTimerInfo) {
         timePoint = info->timer.now() + info->timerOffset
                   - kj::origin<kj::TimePoint>() + kj::UNIX_EPOCH;
       }
+      // We're at script startup time -- just return the Epoch.
     }
+    return (timePoint - kj::UNIX_EPOCH) / kj::MILLISECONDS;
+  }
 
-    // If we're on neither a request- nor inspector-serving thread, then we're at script startup
-    // time -- just return the Epoch.
+  void setInspectorTimerInfo(kj::Timer& timer, kj::Duration timerOffset) {
+    auto lockedState = state.lockExclusive();
+    lockedState->inspectorTimerInfo = InspectorTimerInfo { timer, timerOffset, getCurrentThreadId() };
+  }
 
-    return (timePoint - kj::UNIX_EPOCH) / kj::MILLISECONDS;
+  void setChannel(Worker::Isolate::InspectorChannelImpl& channel) {
+    auto lockedState = state.lockExclusive();
+    lockedState->channel = channel;
```

Comment:
What if there's an existing connection and so `lockedState->channel` is non-null here? Looks like when the other connection later dies, it'll call `resetChannel()` which will incorrectly clear `lockedState->channel`?

Reply:
Comment and invariant added. I might be misunderstanding, but I don't think this is a problem in practice
due to there only being one connection at a time (https://github.com/cloudflare/workerd/blob/7b97887d6dde83da067caaaac946eb6421d5a36c/src/workerd/io/worker.c%2B%2B#L2521).